rental/models.py

```diff
@@ -18,14 +21,36 @@ def get_image_filename(instance, old_filename):
   )
   return filename
 
+def check_user_info(sender, request, user, **kwargs):
+  user_profile = UserProfile(user = user)
+  response = login(request, *args, **kwargs)
+  if request.user.is_authenticated():
+    messages.info(request, "Welcome auth...")
+    if user_profile.phone_number is None:
+      messages.info(request, "Welcome phone...")
+  return response
 
 
 
-class UserInfo(models.Model):
+class UserProfile(models.Model):
   user = models.OneToOneField(User, on_delete = models.CASCADE)
   phone_regex = RegexValidator(regex = r'^\+?1?\d{9,15}$',
                               message = "Phone number must be entered in the format: '+999999999'. Up to 15 digits allowed.")
-  phone_number = models.CharField(validators = [phone_regex], blank=False, max_length = 15) #validators should be a list
+  phone_number = models.CharField(validators = [phone_regex], blank=True, max_length = 15) #validators should be a list
+
+def create_profile(sender, **kwargs):
+    user = kwargs["instance"]
+    if kwargs["created"]:
+        user_profile = UserProfile(user=user)
+        user_profile.save()
+post_save.connect(create_profile, sender=User)  
+
+def create_profile(sender, **kwargs):
```

Comment:
fixed duplicate. and yes its django signal that gets generated everytime a new user is created